app/frontend/src/common/authenticate.js

```diff
@@ -62,15 +62,28 @@ export default {
     localStorage.removeItem('idToken')
     ApiService.authHeader()
   },
-  setTokenExpireAction: function (instance) {
-    instance.onTokenExpired = () => {
-      instance.updateToken().success(() => {
-        this.setLocalToken(instance)
-      }).error(() => {
-        this.removeLocalToken()
-        instance.clearToken()
-      })
-    }
+
+  scheduleRenewal (instance) {
+    const expiresAt = instance.tokenParsed.exp
+
+    // set delay to 50% of token's lifespan (in seconds), or at least 60 seconds
+    let delay = (expiresAt - (Date.now() / 1000)) * 0.5
+    delay = delay > 60 ? delay : 60
+
+    setTimeout(() => {
+      this.renewToken(instance)
+    }, delay * 1000)
+  },
+
+  renewToken (instance) {
+    instance.updateToken(1800).success(() => {
+      this.setLocalToken(instance)
```

Comment:
On updateToken success, it actually returns "refreshed" (true/false) - so I think we only actually need to call setLocalToken if that has happened.